onnx/test/shape_inference_test.py

```diff
@@ -288,31 +317,52 @@ def test_transpose_partial(self, _, version) -> None:
         )
 
     @parameterized.expand(all_versions_for("Transpose"))
-    def test_transpose_preexisting_incorrect_shape(self, *_) -> None:
+    def test_transpose_preexisting_incorrect_shape(self, _, version) -> None:
         graph = self._make_graph(
             [("X", TensorProto.FLOAT, (2, 3, 4))],
             [make_node("Transpose", ["X"], ["Y"], perm=[1, 0, 2])],
             [make_tensor_value_info("Y", TensorProto.FLOAT, (5, 5, 5))],
+            version,
         )
-        self.assertRaises(onnx.shape_inference.InferenceError, self._inferred, graph)
+        if version == 1:
```

Comment:
I hope we have less logic inside a test (avoid if), but at the same time I don’t have a good alternative here

Reply:
May be define a utility function like
```py
def InferenceErrorFor(version):
   return RuntimeError if (version == 1) else InferenceError
```
and use it?